x/profile/keeper/msg_server_update_coordinator_description_test.go

```diff
@@ -0,0 +1,83 @@
+package keeper
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/spn/testutil/sample"
+	"github.com/tendermint/spn/x/profile/types"
+)
+
+func TestMsgUpdateCoordinatorDescription(t *testing.T) {
+	var (
+		addr        = sample.AccAddress()
+		msgCoord    = msgCreateCoordinator()
+		ctx, k, srv = setupMsgServerAndKeeper(t)
+		wCtx        = sdk.WrapSDKContext(ctx)
+	)
+	if _, err := srv.CreateCoordinator(wCtx, &msgCoord); err != nil {
+		t.Fatal(err)
+	}
+	tests := []struct {
+		name string
+		msg  types.MsgUpdateCoordinatorDescription
+		err  error
+	}{
+		{
+			name: "not found address",
+			msg: types.MsgUpdateCoordinatorDescription{
+				Address: addr,
+			},
+			err: sdkerrors.Wrap(types.ErrCoordAddressNotFound, addr),
+		}, {
+			name: "update one value",
+			msg: types.MsgUpdateCoordinatorDescription{
+				Address: msgCoord.Address,
+				Description: &types.CoordinatorDescription{
+					Identity: "update",
+				},
+			},
+		}, {
+			name: "update all values",
+			msg: types.MsgUpdateCoordinatorDescription{
+				Address: msgCoord.Address,
+				Description: &types.CoordinatorDescription{
+					Identity: "update",
+					Website:  "update",
+					Details:  "update",
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := srv.UpdateCoordinatorDescription(wCtx, &tt.msg)
+			if tt.err != nil {
+				require.ErrorIs(t, err, tt.err)
+				return
+			}
+			require.NoError(t, err)
+			coordByAddr, found := k.GetCoordinatorByAddress(ctx, tt.msg.Address)
+			assert.True(t, found, "coordinator by address not found")
+
+			coord := k.GetCoordinator(ctx, coordByAddr.CoordinatorId)
+			assert.True(t, found, "coordinator id not found")
+			assert.EqualValues(t, tt.msg.Address, coord.Address)
+			assert.EqualValues(t, coordByAddr.CoordinatorId, coord.CoordinatorId)
+
+			if len(tt.msg.Description.Identity) > 0 {
```

Comment:
We should add a condition that the value remain unchanged if the field in the message is empty.
It would imply fetching the value and parsing the fields before sending the message with `srv.UpdateCoordinatorDescription(wCtx, &tt.msg)`

```
if tt.err != nil {
  coordByAddr, found := k.GetCoordinatorByAddress(ctx, tt.msg.Address)
  coord := k.GetCoordinator(ctx, coordByAddr.CoordinatorId)
  oldIdentity, oldWebsite,... := coord.Identity, coord.Website, ...
}
```

Reply:
great, nice catch. I made an implementation based on your suggestion